src/app/icd/ICDManager.h

```diff
@@ -33,6 +33,14 @@ namespace app {
 // Used in unit tests
 class TestICDManager;
 
+// This structure is used for the creation an ObjectPool of ICDStateObserver pointers
+struct ObserverPointer
```

Comment:
Should this be inside the `ICDManager` class

Reply:
Just not a fan for it to be global to the chip::app namespcae... I think it should be in an icd namespace or class.

Reply:
Why so? The ICDStateObserver isn't in an ICD namespace nor in the ICDManager class